nservicebus/outbox/index.md

```diff
@@ -79,8 +79,61 @@ In order to gradually convert an entire system from the DTC to the Outbox:
 WARNING: When verifying outbox functionality, it can sometimes be helpful to temporarily [stop the MSDTC Service](https://technet.microsoft.com/en-us/library/cc770732.aspx). This ensures that the Outbox is working as expected, and no other resources are enlisting in distributed transactions.
 
 
+## Message identity and idempotent processing
+
+If handlers are idempotent then outbox might not be required at all. When considering not using the outbox and implementing idempotent processing keep in mind that any generated message *must* apply the same message identifier if invoked more-then-once to have deduplication working on the infrastructure level or to solve this at the logical level by validating message data. [Read more on how to set a message identifier.](/nservicebus/messaging/message-identity.md)
```

Comment:
> When considering not using the outbox and implementing idempotent processing keep in mind that any generated message *must* apply the same message identifier if invoked more-then-once to have deduplication working on the infrastructure level or to solve this at the logical level by validating message data.

I'm still not fully understanding this sentence. Why do you have to set the same message identifier when not using the outbox? What infrastructure is doing deduplication?

Reply:
> Why do you have to set the same message identifier when not using the outbox?

If you want to dedupe on the infrastructure level or on the endpoint level based on message identity. All outgoing messages must be the same with or without outbox.

> What infrastructure is doing deduplication?